modules/tinymce/src/core/main/ts/lookup/UserLookup.ts

```diff
@@ -0,0 +1,174 @@
+import { Obj, Type, Arr } from '@ephox/katamari';
+
+import Editor from '../api/Editor';
+import * as Options from '../api/Options';
+
+/**
+ * TinyMCE User Lookup API
+ * Handles user information retrieval and caching.
+ *
+ * @class tinymce.UserLookup
+ * @example
+ * // Retrieve the current user ID
+ * tinymce.activeEditor.userLookup.getCurrentUserId();
+ *
+ * // Fetch user information by IDs which returns array of promises
+ * const promises = tinymce.activeEditor.userLookup.fetchUsers(['user-1', 'user-2']);
+ * Promise.all(promises).then((users) => {
+ *   users.forEach(user => console.log('User found:', user));
+ * }).catch((error) => {
+ *   console.error('Error fetching users:', error);
+ * });
+ */
+
+type UserId = string;
+
+export interface User {
+  id: UserId;
+  name?: string;
+  avatar?: string;
+  description?: string;
+  custom?: Record<string, any>;
+}
+
+export interface UserLookup {
+  /**
+   * Retrieves the current user ID from the editor.
+   *
+   * @method getCurrentUserId
+   * @return {string} The current user ID.
+   */
+  getCurrentUserId: () => UserId;
+
+  /**
+   * Fetches user information using a provided array of userIds.
+   *
+   * @method fetchUsers
+   * @param {string[]} userIds - A list of user IDs to fetch information for.
+   * @throws {Error} Throws an error if any of the user IDs are invalid or if the fetch fails.
+   * @return {Promise<User>[]} A promise that resolves to an array of users and information about them.
+   */
+  fetchUsers: (userIds: UserId[]) => Promise<User>[];
+}
+
+const isUserObject = (val: unknown): val is User =>
+  Type.isObject(val)
+  && val.hasOwnProperty('id')
+  && Type.isNonNullable((val as User).id) && Type.isString((val as User).id as UserId)
+  && !(Type.isNonNullable((val as User).name) && !Type.isString((val as User).name))
+  && !(Type.isNonNullable((val as User).avatar) && !Type.isString((val as User).avatar))
+  && !(Type.isNonNullable((val as User).description) && !Type.isString((val as User).description));
+
+const validateResponse = (items: unknown): User[] => {
+  if (!Array.isArray(items)) {
+    throw new Error('fetch_users must return an array');
+  }
+
+  if (items.length === 0) {
+    return [];
+  }
+
+  const users = Arr.filter(items, (item) => isUserObject(item));
+  return users;
+};
+
+const isValidUserId = (userId: unknown): userId is UserId =>
+  Type.isString(userId)
+  && userId.length > 0;
+
+const UserLookup = (editor: Editor): UserLookup => {
+  const userCache: Record<UserId, Promise<User>> = {};
```

Comment:
You used a `Map` for the pending resolvers, which is good, but perhaps we should use one for the cache as well?